multi_physics/QED/include/picsar_qed/containers/picsar_tables.hpp

```diff
@@ -24,6 +24,65 @@ namespace picsar{
 namespace multi_physics{
 namespace containers{
 
+    //________________ Auxiliary functions _____________________________________
+
+    // These auxiliary functions provide a way to check if an object of type
+    // VectorType has a pxr_sync() method and call such method if this is the case.
+    // __aux_sync_vec(VectorType& vec) calls the right version of __aux_imp_sync_vec
+    // (the one which does nothing or the one which calls vec.pxr_sync()) thanks to
+    // SFINAE (Substitution Failure Is Not An Error) and overload resolution rules.
+
+    /**
+    * This function is called if VectorType has a pxr_sync() method.
+    * If VectorType doesn't have a pxr_sync() method, substitution of the
+    * template parameter in decltype(vec.pxr_sync(), void()) fails.
+    * Therefore, this function specialization is discarded.
+    * Otherwise, this function overload has priority over the other one below,
+    * since __aux_imp_sync_vec is called by __aux_sync_vec with an int as a
+    * second argument. Therefore, this specialization is a better match.
+    *
+    * @tparam VectorType the vector type to be used (e.g. std::vector<double>)
+    * @param vec a reference to a VectorType
+    * @param unused
+    */
+    template<typename VectorType>
+    inline auto
+    __aux_imp_sync_vec(VectorType& vec, int)
```

Comment:
Note that `__` is reserved in the C++ std and might even be UB.

Instead, maybe just name them normally and wrap such things in a `details` namespace.

Reply:
You are right, sorry... I've just followed your suggestion